### PR TITLE
Skip suppressed optional arguments on tcsh

### PR DESCRIPTION
**Pull request by simaoafonso-pwt**
Followup #57.

@casperdcl Sorry about all the churn, I guess I didn't test hard enough.

### CODE REVIEW

File: shtab/__init__.py [hunk 1]
```diff
@@ -624,14 +624,15 @@ def recurse_parser(cparser, positional_idx, requirements=None):
...
             log.debug('%s| Optional: %s', log_prefix, optional.dest)
-            # Mingle all optional arguments for all subparsers
-            for optional_str in optional.option_strings:
-                log.debug('%s| | %s', log_prefix, optional_str)
-                if optional_str.startswith('--'):
-                    optionals_double.add(optional_str[2:])
-                elif optional_str.startswith('-'):
-                    optionals_single.add(optional_str[1:])
-                specials.extend(get_specials(optional, 'n', optional_str))
+            if optional.help == SUPPRESS:
```
Comment (casperdcl) [inline shtab/__init__.py:627]:
```suggestion
            if optional.help != SUPPRESS:
```

unless I'm missing something? :sweat_smile:

Comment (simaoafonso-pwt) [reply shtab/__init__.py]: Oops. I toyed with

```python
if optional.help == SUPPRESS:
   continue
```

**Review by casperdcl — APPROVED** [review]
tested & fixed example script - lgtm now

**Pull request merged**